code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,227 @@
+#define TARGET_INDEX 1
+#define IDENTIFIER_INDEX 2
+#define START_TIME_INDEX 3
+#define REQUEST_INDEX 4
+#define MESSAGE_INDEX 5
+#define EXTRA_TARGETS_INDEX 6
+#define LANGUAGE_INDEX 7
+#define LOCAL_INDEX 8
+#define MESSAGE_RANGE_INDEX 9
+
+SUBSYSTEM_DEF(tts)
+	name = "Text To Speech"
+	wait = 0.05 SECONDS
+	priority = FIRE_PRIORITY_TTS
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	/// Queued HTTP requests that have yet to be sent
+	var/list/queued_tts_messages = list()
+
+	/// HTTP requests currently in progress but not being processed yet
+	var/list/in_process_tts_messages = list()
+
+	/// HTTP requests that are being processed to see if they've been finished
+	var/list/current_processing_tts_messages = list()
+
+	/// A list of available speakers
+	var/list/available_speakers = list()
+
+	var/list/cached_voices = list()
+
+	/// Whether TTS is enabled or not
+	var/tts_enabled = FALSE
```

Comment:
so config cannot be swapped midround? that seems potentially really annoying

Reply:
it's the fact that TTS connection needs to happen at the start of the round or it won't work properly because it uplinks with a html server.

Reply:
so we couldn't make that connection later? or reattempt it if variables change?

Reply:
maybe, but it wouldn't work at all since the voices have already been cached in preferences. It generally just breaks in general if you try loading it midround and some mobs will have voices, some will not, etc. It'd be possible to let them turn it off midround, but there's too many issues with trying to enable it midround.

Reply:
that's depressing but understandable. I would kinda like to use a signal system to make it function but I get that's not totally possible